kale/loaddata/multiomics_datasets.py

```diff
@@ -0,0 +1,461 @@
+            pre_transform,
+            target_pre_transform,
+        )
+
+    def extend_data(self, data: Data) -> Data:
+        """Extend data object by adding additional attributes.
+
+        Args:
+            data (Data): An input data object.
+
+        Returns:
+            Extended data object with additional attributes.
+        """
+        # Add train sample weights to the data object
+        train_labels = torch.argmax(data.y[data.train_idx], dim=1)
+        train_sample_weight = self._get_sample_weight(train_labels)
+        data.train_sample_weight = train_sample_weight
+
+        # Add adjacency matrices to the data object
+        edge_index_train, edge_weight_train = self._get_adjacency_info(data.x[data.train_idx], train=True)
+        adj_t_train = SparseTensor(row=edge_index_train[0], col=edge_index_train[1], value=edge_weight_train)
+
+        edge_index, edge_weight = self._get_adjacency_info(
+            data.x[data.train_idx], test_data=data.x[data.test_idx], train=False
+        )
+        adj_t = SparseTensor(row=edge_index[0], col=edge_index[1], value=edge_weight).t()
+
+        data.edge_index = edge_index
+        data.edge_weight = edge_weight
+        data.adj_t = adj_t
+        data.edge_index_train = edge_index_train
+        data.edge_weight_train = edge_weight_train
+        data.adj_t_train = adj_t_train
+
+        return data
+
+    def _get_adjacency_info(
+        self,
+        train_data: torch.Tensor,
+        test_data: torch.Tensor = None,
+        train: bool = True,
+        eps: float = 1e-8,
+        metric: DistanceMetric = DistanceMetric.COSINE,
+    ) -> Tuple:
+        """Calculate sparse adjacency matrix of the input dataset defined by edge indices and edge attributes.
+
+        Args:
+            train_data (torch.Tensor): The training data.
+            test_data (torch.Tensor, optional): The test data. If ''None'' then the adjacency matrix is only calculated
+                on the train data. (default: ``None``)
+            train (bool, optional): Whether to use only train data to calculate the adjacency matrix. If ''False'' then
+                the entire dataset (training and test sets) is used. (default: ``True``)
+            eps (float, optional): Small value to avoid division by zero. (default: 1e-8)
+            metric (DistanceMetric, optional): The metric to compute distance between input matrices. (default:
+                ``DistanceMetric.COSINE``)
+
+        Returns:
+            A tuple of edge indices and edge attributes.
+        """
+        num_train = train_data.shape[0]
+        num_test = 0 if test_data is None else test_data.shape[0]
+        if train:
+            adj_matrix = calculate_distance(train_data, eps=eps, metric=metric)
+            self._find_sim_threshold(adj_matrix, num_train)
+            non_zero_entries = self._generate_sparse_adj(adj_matrix, self_loop=train)
+            adj_matrix = torch.mul(adj_matrix, non_zero_entries)
+        else:
```

Comment:
Is else test or validation?

Reply:
"train" is an input parameter of the method, and I have included a description for this parameter in the docstring for clarity:

"train (bool, optional): Whether to use only train data to calculate the adjacency matrix. If ''False'' then the entire dataset (training and test sets) is used. (default: ``True``)"